client/views/voip/VoIPLayout.tsx

```diff
@@ -44,6 +47,78 @@ class VoIPLayout
 		this.webSocketPath = React.createRef();
 		this.callTypeSelection = React.createRef();
 		this.logger = new Logger('VoIPLayout');
+		this.extensionConfig = null;
+	}
+
+	private async initUserAgent(): Promise<void> {
+		let extension = '80000';
+		if (this.userName.current && this.userName.current.value) {
+			extension = this.userName.current.value;
+		}
+		try {
+			this.extensionConfig = await APIClient.v1.get('connector.extension.getRegistrationInfo', {
+				extension,
+			});
+			this.logger?.info('list = ', JSON.stringify(this.extensionConfig));
+		} catch (error) {
+			this.logger?.error('error in API');
+			throw error;
+		}
+
+		if (this.extensionConfig.extension && this.userName.current) {
+			this.userName.current.textContent = this.extensionConfig.extension;
+			this.userName.current.disabled = true;
+			this.config.authUserName = this.extensionConfig.extension;
+		}
+		if (this.extensionConfig.password && this.password.current) {
+			this.password.current.value = this.extensionConfig.password;
+			this.password.current.disabled = true;
+			this.config.authPassword = this.extensionConfig.password;
+		}
+		if (this.extensionConfig.sipRegistrar && this.registrar.current) {
+			this.registrar.current.value = this.extensionConfig.sipRegistrar;
+			this.registrar.current.disabled = true;
+			this.config.sipRegistrarHostnameOrIP = this.extensionConfig.sipRegistrar;
+		}
+		if (this.extensionConfig.websocketUri && this.webSocketPath.current) {
+			this.webSocketPath.current.value = this.extensionConfig.websocketUri;
+			this.webSocketPath.current.disabled = true;
+			this.config.webSocketURI = this.extensionConfig.websocketUri;
+		}
+
+		this.config.enableVideo = this.state.enableVideo;
+		this.config.connectionDelegate = this;
+		this.config.iceServers = [{ urls: 'stun:stun.l.google.com:19302' }];
```

Comment:
![image](https://user-images.githubusercontent.com/34130764/135212509-5ef0aae2-decd-437c-baf9-ea2e15949892.png)
Hi, @amolghode1981 I was wondering if instead of hardcoding these values, can we instead use the following server settings we already have?

Reply:
Oh, I am not aware of this setting. Thanks for pointing this out to me. 
Yes. This hardcoding is temporary to get things testable and verifiable. I am now wondering why there is no Figma for this particular setting. Anyhow this needs a discussion. 
Are these settings available to the agent or only for the administrator?  If they are available for only administrator, we may have to add this component. These settings ideally are done by the agent. 

I am resolving this for now. But I will add a comment in the code to mention this concern.